contracts/test/LONStaking/Setup.t.sol

```diff
@@ -0,0 +1,100 @@
+    uint256 constant BPS_RAGE_EXIT_PENALTY = 500;
+    uint256 constant BPS_MAX = 10000;
+
+    uint256 userPrivateKey = uint256(1);
+    uint256 otherPrivateKey = uint256(2);
+
+    address user = vm.addr(userPrivateKey);
+    address other = vm.addr(otherPrivateKey);
+    address upgradeAdmin = address(0x133701);
+    address spender = address(0x133702);
+    address fuzzingUserStartAddress = address(0x133703);
+
+    Lon lon = new Lon(address(this), address(this));
+    xLON xLon;
+    LONStaking lonStaking;
+
+    uint256 DEFAULT_STAKE_AMOUNT = 1e18;
+    uint256 DEADLINE = block.timestamp + 1;
+
+    // effectively a "beforeEach" block
+    function setUp() public virtual {
+        // Setup
+        LONStaking lonStakingImpl = new LONStaking();
+        bytes memory initData = abi.encodeWithSignature(
+            "initialize(address,address,uint256,uint256)",
+            address(lon), // LON
+            address(this), // Owner
+            COOLDOWN_IN_DAYS,
+            BPS_RAGE_EXIT_PENALTY
+        );
+
+        xLon = new xLON(address(lonStakingImpl), upgradeAdmin, initData);
+        lonStaking = LONStaking(address(xLon));
+
+        // Deal 100 ETH to user
+        deal(user, 100 ether);
+        // Mint LON to user
+        lon.mint(user, 100 * 1e18);
+        // User approve LONStaking
+        vm.prank(user);
+        lon.approve(address(lonStaking), type(uint256).max);
+
+        // Label addresses for easier debugging
+        vm.label(user, "User");
+        vm.label(address(this), "TestingContract");
+        vm.label(address(lon), "LONContract");
+        vm.label(address(xLon), "xLONContract");
+        vm.label(address(lonStakingImpl), "LONStakingImplementationContract");
+        vm.label(address(lonStaking), "LONStakingContract");
+    }
+
+    /*********************************
+     *          Test Helpers         *
+     *********************************/
+
+    function _getEIP712Hash(bytes32 structHash) internal view returns (bytes32) {
+        string memory EIP191_HEADER = "\x19\x01";
+        bytes32 DOMAIN_SEPARATOR = lonStaking.DOMAIN_SEPARATOR();
+        return keccak256(abi.encodePacked(EIP191_HEADER, DOMAIN_SEPARATOR, structHash));
+    }
+
+    function _getExpectedXLON(uint256 stakeAmount) internal view returns (uint256) {
+        uint256 totalLon = lon.balanceOf(address(lonStaking));
+        uint256 totalShares = lonStaking.totalSupply();
+        if (totalShares == 0 || totalLon == 0) {
+            return stakeAmount;
+        } else {
+            return stakeAmount.mul(totalShares).div(totalLon);
+        }
+    }
+
+    function _stake(address staker, uint256 stakeAmount) internal {
+        vm.startPrank(staker);
+        if (lon.allowance(staker, address(lonStaking)) == 0) {
+            lon.approve(address(lonStaking), type(uint256).max);
+        }
+        lonStaking.stake(stakeAmount);
+        vm.stopPrank();
+    }
+
+    function simulateBuyback(uint256 amount) internal {
```

Comment:
Shall we prefix underscore to this internal function too?

Reply:
Yes! Its original without underscore, but it should need to be added.

Reply:
So is this PR ready or need adjusted?

Reply:
Sorry, I have already pushed a new commit!
Waiting for the CI program...